### PR TITLE
Anki mode (Fixes #347)

### LINKED ISSUE #86: Close but no cigar & disable partial answer match
Currently it's not possible to turn off the partial matching of answers (there is an option to disable cheating but i still want to have the ignore option).

My ideal would be behaviour like the "close but no cigar" user script, where a partial correct answer notifies you that you are close, but doesn't accept the answer yet and waits for the 100% correct match.

### LINKED ISSUE #347: Feature Request: Anki mode
A special mode that works like Anki. Tapping on the screen reveals the answer (reading or meaning depending on the question but not both - to avoid spoiling the following questions). On the answer screen there would also be "Correct" / "Incorrect" buttons that marks the answer (or even maybe Tinder-style swipe left / right). 

This would basically be equivalent of submitting an incorrect answer, then hitting "My Answer Was Correct", but without the reveal of the full answer card and with an optimized UI intended for this interaction. This would definitely speed up reviews as it is a burden to type the answers every time.

### PR DESCRIPTION
**Pull request by UInt2048**
Fixes #347 by implementing Anki mode

### CODE REVIEW

**Comment by UInt2048** [comment]
Here's a video demoing all the new features: https://i.imgur.com/rwXG6EV.mp4

**Comment by UInt2048** [comment]
@davidsansome Ready for review

**Comment by UInt2048** [comment]
@davidsansome Is there anything else you need from me before you merge this?

I'd be happy to help with whatever it is...

**Comment by UInt2048** [comment]
The remaining former contents of this pull request has been split into multiple PRs.

* This PR fixes Anki mode.
* Editing notes has been migrated to #479 
* Pausing on partially correct answers has been migrated to #482 
* Exact match on burns only isn't being requested, because some other PRs really need to be merged first.

**Comment by UInt2048** [comment]
The remaining former contents of this pull request has been split into multiple PRs.

* This PR fixes Anki mode.
* Editing notes has been migrated to #479 
* Pausing on partially correct answers has been migrated to #482 
* Exact match on burns only isn't being requested, because some other PRs really need to be merged first.

@davidsansome This PR and the PRs just created today are ready for review.

File: ios/ReviewViewController.swift [hunk 1]
```diff
@@ -1011,6 +1017,13 @@ class ReviewViewController: UIViewController, UITextFieldDelegate, SubjectDelega
   }
 
   func submit() {
+    if Settings.ankiMode {
+      // Mark the answer incoorect to show the details. This can still be overriden.
```
Comment (davidsansome) [inline ios/ReviewViewController.swift:1021]: nit: s/incoorect/incorrect/

**Review by davidsansome — APPROVED** [review]
This is awesome! Thanks for splitting it up into separate PRs!

File: ios/ReviewViewController.swift (hunk 1 above)
Comment (UInt2048) [reply ios/ReviewViewController.swift]: Thanks! I'm not sure how that typo escaped me.

**Pull request merged**